internal/api/external.go

```diff
@@ -558,6 +558,8 @@ func (a *API) Provider(ctx context.Context, name string, scopes string) (provide
 		return provider.NewSpotifyProvider(config.External.Spotify, scopes)
 	case "slack":
 		return provider.NewSlackProvider(config.External.Slack, scopes)
+	case "slack_legacy":
+		return provider.NewSlackLegacyProvider(config.External.Slack, scopes)
```

Comment:
this wouldn't be backward compatible for existing legacy oauth apps right since they will need to update their endpoint to point to `/authorise?provider=slack_legacy` instead of `/authorise?provider=slack` 

i would prefer to rename the new provider to `slack_oidc` like what we did for the linkedin provider